packages/react-components/src/shared/src/slots.tsx

```diff
@@ -52,15 +52,18 @@ export function useRawSlots(children: ReactNode, slots: string[]) {
     return useMemo(() => getRawSlots(children, slots), [children, slots]);
 }
 
-interface Slots {
+interface RequiredDefaultSlot {
     _: {
         defaultWrapper?: ComponentType;
         required?: string[];
-    },
-    [x: string]: any;
+    }
 }
 
-export function getSlots(children: ReactNode, { _ = {}, ...slots }: Slots) {
+type SlotElements<T extends RequiredDefaultSlot> = {
```

Comment:
given
```
    const slots} = useSlots(children, useMemo(() => ({
        _: {
            defaultWrapper: Content
        },
        icon: {
            className: "o-ui-alert-icon"
        },
        content: {
            className: "o-ui-alert-content",
            as: AlertContent
        },
        button: {
            variant: "ghost",
            color: "inherit",
            condensed: true,
            size: "sm",
            className: "o-ui-alert-action"
        }
    }), []));
```

before : 
slot was type with [key:string]: any

now : 
slot is typed with 
```
{
    icon?:ReactElement;
    content?:ReactElement;
    button?: ReactElement;
   [key:string]: ReactElement
}
```
Wich means we give suggestion for autocomplete  while destructuring + allow any other slot that would be extracted (to support the defaultWrapper case)